tests/child_functions.rs

```diff
@@ -150,3 +150,41 @@ fn call() {
     let mut runtime = RootRuntime::new(&code, &[], [0u8; 32]);
     runtime.execute();
 }
+
+#[cfg(all(test, feature = "debug"))]
+#[cfg_attr(feature = "debug", test)]
+fn print() {
+    let child_code = r#"
+    (module
+        (import
+            "env"
+            "eth2_return"
+            (func
+                $eth2_return
+                (param i32)
+                (param i32)
+                (result i32)))
+        (import "env" "print" (func $print (param i32) (param i32)))
+        (memory (export "memory") 1)
+        (data (i32.const 0) "hello world")
+        (func $main (export "main") (result i32) (local $x i32)
+            (; print data ;)
+            (call $print (i32.const 0) (i32.const 11))
+
+            (; Return a value to the caller ;)
+            (i32.store (i32.const 10) (i32.const 4321))
+            (call $eth2_return (i32.const 10) (i32.const 4))
+
+        )
+    )
+    "#;
+
+    let code = compile_wat(child_code);
+    let mut runtime = RootRuntime::new(&code, &[], [0u8; 32]);
+
+    runtime.set_logger(Box::new(|b| {
+        assert_eq!(b, "hello world");
```

Comment:
The bad thing about this test is that if for some reason `print` *isn't* called, the closure will not execute and it will succeed with a false positive. Any thoughts on how I can ensure the closure is executed?

Reply:
Create a `Rc<Cell<bool>>`, clone it into the closure and set it, and check outside.